projects/ccf-rui/src/app/shared/components/visibility-toggle/visibility-toggle.component.html

```diff
@@ -0,0 +1,10 @@
+<div (click)="toggleVisibility()" class="toggle-wrapper">
+    <div class="visibility-icon hidden" *ngIf="!visible; else visibleIcon">
```

Comment:
Why not use regular binding? `<mat-icon ... [svgIcon]="visible ? 'app:visibility_on' : 'app:visibility_off'">`

Reply:
You could even create a getter in the class and then simply bind `<mat-icon ... [svgIcon]="icon">`